spec-tests/src/test/java/org/candlepin/spec/imports/DerivedProductImportSpecTest.java

```diff
@@ -0,0 +1,157 @@
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+        OwnerDTO distOwner = ownerApi.createOwner(Owners.random());
+        ApiClient distUserClient = ApiClients.basic(UserUtil.createUser(client, distOwner));
+
+        ProductDTO derivedProduct = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .attributes(List.of(
+            ProductAttributes.Cores.withValue("6"),
+            ProductAttributes.Sockets.withValue("8"))));
+
+        ProductDTO stackedDatacenterProduct = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .attributes(List.of(
+            ProductAttributes.VirtualLimit.withValue("unlimited"),
+            ProductAttributes.StackingId.withValue("mixed-stack"),
+            ProductAttributes.Sockets.withValue("2"),
+            ProductAttributes.MultiEntitlement.withValue("yes")))
+            .derivedProduct(derivedProduct));
+
+        PoolDTO datacenterPool = ownerApi.createPool(owner.getKey(),
+            Pools.random(stackedDatacenterProduct)
+            .contractNumber("222")
+            .accountNumber("")
+            .orderNumber(""));
+
+        // create the distributor consumer
+        ConsumerDTO distributor = userClient.consumers().createConsumer(
+            Consumers.random(owner)
+            .type(ConsumerTypes.Candlepin.value())
+            .facts(Map.of("distributorVersion", "sam-1.3")));
+        ApiClient distributorClient = ApiClients.ssl(distributor);
+        String distVersion = StringUtil.random("version");
+        client.distributorVersions().create(new DistributorVersionDTO()
+            .name(distVersion)
+            .displayName("SAM")
+            .capabilities(Set.of(new DistributorVersionCapabilityDTO().name("cert_v3"),
+                new DistributorVersionCapabilityDTO().name("derived_product"))));
+        distributorClient.consumers().updateConsumer(distributor.getUuid(),
+            new ConsumerDTO().facts(Map.of("distributor_version", distVersion)));
+        // entitlements from data center pool
+        distributorClient.consumers().bindPool(distributor.getUuid(), datacenterPool.getId(), 10);
+
+        // make manifest
+        File export = client.consumers().exportData(distributor.getUuid(), null, null, null);
+        export.deleteOnExit();
+        // remove client at 'host'
+        userClient.consumers().deleteConsumer(distributor.getUuid());
+        client.pools().deletePool(datacenterPool.getId());
+        // import to make org at 'distributor'
+        AsyncJobStatusDTO importJob = client.owners().importManifestAsync(distOwner.getKey(), List.of(), export);
+        assertEquals("FINISHED", client.jobs().waitForJob(importJob).getState());
+
+        List<PoolDTO> pools = ownerApi.listOwnerPoolsByProduct(distOwner.getKey(),
+            stackedDatacenterProduct.getId());
+        assertThat(pools).hasSize(1);
```

Comment:
You can use `assertThat(pools).singleElement();` that will also verify that pools is not null.